File: observablecollection/src/main/java/sk/teamsoft/observablecollection/SimpleAdapterSource.java
package sk.teamsoft.observablecollection;

import android.support.annotation.LayoutRes;

import java.util.List;

/**
 * Simple {@link ObservableAdapter} Source implementation which requires only one view type
 * and therefore only one view layout
 * @author Dusan Bartos
 */
@SuppressWarnings("WeakerAccess")
public final class SimpleAdapterSource<T> extends AdapterSource<T> {
    @LayoutRes private final int layout;

    /**
     * Creates simple source for {@link ObservableAdapter} with just single view type
     * Uses 'true' flag for resolving movement in DiffUtil
     * @param data   default data set
     * @param layout viewholder layout resource
     */
    public SimpleAdapterSource(List<T> data, @LayoutRes int layout) {
        this(data, layout, false);
    }

    /**
     * Creates simple source for {@link ObservableAdapter} with just single view type
     * @param data           default data set
     * @param layout         viewholder layout resource
     * @param detectMovement false to disable movement detection during updates (for better performance)
     */
    public SimpleAdapterSource(List<T> data, @LayoutRes int layout, boolean detectMovement) {
        super(data, detectMovement);
        this.layout = layout;
    }

    @Override protected int getViewType(int pos) {
        return -1;
    }

    @Override public int getLayout(int viewType) {
        return layout;
    }
}

File: observablecollection/src/main/java/sk/teamsoft/observablecollection/SimpleAdapterSource.kt
package sk.teamsoft.observablecollection

import androidx.annotation.LayoutRes

/**
 * Simple [ObservableAdapter] Source implementation which requires only one view type
 * and therefore only one view layout

 * @author Dusan Bartos
 */
class SimpleAdapterSource<T> constructor(
    data: List<T>,
    @LayoutRes private val layout: Int,
    detectMove: Boolean = false
) : AdapterSource<T>(data, detectMove) {

    constructor(data: List<T>, @LayoutRes layout: Int) : this(data, layout, false)
    constructor(@LayoutRes layout: Int) : this(emptyList(), layout, false)

    override fun getViewType(pos: Int) = -1
    override fun getLayout(viewType: Int) = layout
}
